### ISSUE TITLE
noImplicitAny option do not generate an error when the function generating this coercicion is passed as a function parameter.

### ISSUE THREAD

**Issue by sledorze** [Microsoft/TypeScript#8773]
**TypeScript Version:** 

1.9.0-dev.20160523-1.0

**Code** (with noImplicitAny option turned ON)

```ts

class Foo {

  identity<T>(x: T): T {
    return x
  }

  bar<T>(g: (a: T) => T): (a: T) => T {
    return g;
  }
  
  constructor() {
    let v = a => this.identity(a)
    let w = this.bar(a => this.identity(a))
  }
}

```

**Expected behavior:**

The line binding an expression to w should error stating that 'a' implicitely has an 'any' type
OR (better)
v and w should be polymorphic (and have type <T>(a:T) => T

**Actual behavior:** 

v and w both have type (a:{}) => {} AND only the expression of v generates an error stating that 'a' implicitly has an 'any' type. (noImplicitAny option turned on)

**Comment by mhegazy** [comment]
The current behavior looks right to me.
- `a` in `let v = a => this.identity(a)` implicitly has the type `any`,as there is no type annotation or a contextual type.
- `v` does not have the type `any`, it is `(a:any) => any`, and it is not implicit. so no error should be reported here.
- `a` in `this.bar(a => this.identity(a))` has a contextual type, that is the result of the inference for T, which happens to get no useful type, so falls back to the `{}`. and `a` gets that. again nothing implicit about this.  see https://github.com/Microsoft/TypeScript/issues/5254 for this.
- `w` does not have the type `any`, it is `(a:{}) => {}`, and it is not implicit. so no error should be reported here.

i think https://github.com/Microsoft/TypeScript/issues/5254 is the real culprit and not implictAny warnings.
